### PR TITLE
UF-4531: Implement notification logic.

### PR DESCRIPTION
**Pull request by foobar-1337**

### CODE REVIEW

File: src/main/java/se/sundsvall/seabloader/service/InvoiceService.java [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
 	private static final Logger LOGGER = LoggerFactory.getLogger(InvoiceService.class);
-	private static final Status[] STATUSES_OF_INVOICES_TO_SEND  = {UNPROCESSED, EXPORT_FAILED};
+	private static final Status[] STATUSES_OF_INVOICES_TO_SEND = { UNPROCESSED, EXPORT_FAILED };
 
 	@Autowired
 	private InvoiceRepository invoiceRepository;
 
-	@Autowired InvoicePdfMerger invoicePdfMerger;
+	@Autowired
+	InvoicePdfMerger invoicePdfMerger;
```
Comment (josc04) [inline src/main/java/se/sundsvall/seabloader/service/InvoiceService.java:33]: private?

**Review by MickeWallner — DISMISSED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -68,11 +74,12 @@ resilience4j.circuitbreaker.instances.messaging.ignoreExceptions=se.sundsvall.de
...
 # Scheduler settings
-# - To disable a specific schedule, set cron expression to "-" or remove
-#   property for the specific schedule
+# - To disable a specific schedule, set cron expression to "-" 
+#   or remove property for the specific schedule.
 #-------------------------------------------------------------------------------
-scheduler.database.cleaner.cron.expression=0 0 2 * * 7
-scheduler.invoice.exporter.cron.expression=0 0 1 * * *
+scheduler.dbcleaner.cron.expression=0 0 2 * * 7
+scheduler.invoiceexporter.cron.expression=0 0 1 * * *
```
Comment (josc04) [inline src/main/resources/application.properties:81]: Reminder to also change this setting in the property file for production

File: src/main/java/se/sundsvall/seabloader/scheduler/notifier/NotifierSchedulerService.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	}
+
+	/**
+	 * Sends failure notifications to configured recipient.
+	 */
+	private void sendFailureNotification() {
+
+		// Collect data
+		final var failedStatusMap = new EnumMap<Status, Long>(Status.class);
+		Stream.of(Status.values()).forEach(status -> failedStatusMap.put(status, invoiceRepository.countByStatusIn(status)));
```
Comment (josc04) [inline src/main/java/se/sundsvall/seabloader/scheduler/notifier/NotifierSchedulerService.java:70]: What about filtering the Status enum and only include EXPORT_FAILED and IMPORT_FAILED here as there is no need to count rows with other statuses than those? I imagine that you could gain several nanoseconds in execution time that way 😄.

File: src/main/java/se/sundsvall/seabloader/service/InvoiceService.java (hunk 1 above)
Comment (foobar-1337) [reply src/main/java/se/sundsvall/seabloader/service/InvoiceService.java]: Fixed

**Comment by foobar-1337** [comment]
Acted on review comments, fixed broken tests and added feature toggling for the mail notification

**Review by josc04 — APPROVED** [review]

**Pull request merged**